kgtk/graph_embeddings/build_faiss.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+import faiss
+import os
+from tqdm import tqdm
+from kgtk.exceptions import KGTKException
+
+
+# # TODO:
+# - look up what is faiss.set_direct_map_type and if I need more params for this.
+#   - still not sure. Very little information about it is easily available
+# - add support for sharding
+# - add support for cosine similarity as a distance metric
+
+
+def build_faiss(embeddings_file, embeddings_format, no_input_header, index_file_out, index_to_node_file_out,
+                max_train_examples, workers, index_string, metric_type, p=None, verbose=False):
+
+    # validate input file path
```

Comment:
use `KGTKReader` to read the input file as it is an edge file. It handles a lot of exceptions and cases. Take a look at the `cat` command for example.